ocaml/xapi/xapi_host.ml

```diff
@@ -1766,3 +1768,21 @@ let mxgpu_vf_setup ~__context ~host =
 let allocate_resources_for_vm ~__context ~self ~vm ~live =
   (* Implemented entirely in Message_forwarding *)
   ()
+
+let set_iscsi_iqn ~__context ~host ~value =
+  if value = "" then raise Api_errors.(Server_error (invalid_value, ["value"; value]));
+  (* Note, the following sequence is carefully written - see the
+     other-config watcher thread in xapi_host_helpers.ml *)
+  Db.Host.remove_from_other_config ~__context ~self:host ~key:"iscsi_iqn";
+  Db.Host.set_iscsi_iqn ~__context ~self:host ~value;
+  Db.Host.add_to_other_config ~__context ~self:host ~key:"iscsi_iqn" ~value;
```

Comment:
This re-triggers an rpc call to set the iscsi_iqn, thing that we have "kind of" just done above.
Why such a complicated flow?

Reply:
I've added a comment attempting to explain my understanding of it:
+  (* we need to first set the iscsi_iqn field and then the other-config field:
+   * setting the other-config field triggers and update on the iscsi_iqn
+   * field if they are different
+   * *)

Reply:
I think the reason for the complicated behaviour is that we want to keep both fields in sync: when you update iscsi_iqn field we update other-config to match, and when you update other-config we update iscsi_iqn to match. We have to avoid an infinite loop though.

Reply:
Yes, I think it's clearer. Thanks